extension/src/pipeline/util.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  getFileExtension,
+  loadJson,
+  loadCsv,
+  loadTsv,
+  loadYamlAsJs
+} from '../fileSystem'
+import { quickPickOne } from '../vscode/quickPick'
+import { pickFile } from '../vscode/resourcePicker'
+import { Title } from '../vscode/title'
+
+const parseDataFile = (file: string) => {
+  const ext = getFileExtension(file)
+
+  if (ext === '.json') {
+    return loadJson<Record<string, unknown> | unknown[]>(file)
+  }
+
+  if (ext === '.csv') {
+    return loadCsv(file)
+  }
+
+  if (ext === '.tsv') {
+    return loadTsv(file)
+  }
+
+  if (ext === '.yaml') {
+    return loadYamlAsJs<Record<string, unknown>>(file)
+  }
+}
+
+const pickDataFile = () => {
+  return pickFile(Title.SELECT_PLOT_DATA, {
+    filters: {
+      'Data Formats': ['json', 'csv', 'tsv', 'yaml']
+    },
+    openLabel: 'Select'
+  })
+}
+
+const pickTemplateAndFields = async (
+  fields: string[]
+): Promise<{ x: string; y: string; template: string } | undefined> => {
+  const template = await quickPickOne(
+    [
+      'simple',
+      'linear',
+      'confusion',
+      'confusion_normalized',
+      'scatter',
+      'scatter_jitter',
+      'smooth',
+      'bar_horizontal_sorted',
+      'bar_horizontal'
```

Comment:
Let's move this array into `extension/src/cli/dvc/contract.ts` as a constant. Gives us a better chance of catching the need to make updates if/when the list changes in `dvc-render`